Failing live playlist update when lowLatencyMode = false

When using a LL-HLS stream and `lowLatencyMode: false`, hls.js will request part 0 from a future segment. The server can return an error, since it is likely more than 3 parts into the future.

Note that the stream will actually play, since hls.js has some recovery logic, but it means that 3 requests are made for each live playlist update.

### What version of Hls.js are you using?

master / c5698be3404b285cc4c3419f8d297a510b113e09.

### What browser and OS are you using?

### Test stream:

https://stream.sob.m-dn.net/live/sb1-ll/index.m3u8

### Checklist

<!-- Replace [ ] with [x] to check off the list -->

- [x] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [ ] The issue occurs in the stable client on https://hls-js.netlify.com/demo and not just on my page
<!-- The stable client is built from the latest release -->
- [x] The issue occurs in the latest client on https://hls-js-dev.netlify.com/demo and not just on my page
<!-- The latest client is built from the head of the master branch -->
- [x] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [ ] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

1. Play LL-HLS in demo player with `lowLatencyMode: false`.

### Expected behavior

Request future index update using `_HLS_msn=<next-msn>` param, or use regular part updates from `lowLatencyMode: true` logic.

### Actual behavior

Request future index update using `_HLS_msn=<next-msn` & `_HLS_part=0` params, resulting in 503s and multiple requests.

### Console output

```
xhr-loader.ts:121 GET https://stream.sob.m-dn.net/live/sb1-ll/v8Ew9fFqC0AEwfHoR5PS1cF/stream.m3u8?_HLS_msn=307769&_HLS_part=0 503
xhr-loader.ts:184 [error] > 503 while loading https://stream.sob.m-dn.net/live/sb1-ll/v8Ew9fFqC0AEwfHoR5PS1cF/stream.m3u8?_HLS_msn=307769&_HLS_part=0
playlist-loader.ts:600 [warn] > [playlist-loader]: A network error occurred while loading level level: 3 id: 0 group-id: "null"
base-playlist-controller.ts:231 [warn] > [level-controller]: retry playlist loading #1 in 2000 ms after "levelLoadError"
main.js:718 Error event: {type: "networkError", details: "levelLoadError", fatal: false, url: "https://stream.sob.m-dn.net/live/sb1-ll/v8Ew9fFqC0…HoR5PS1cF/stream.m3u8?_HLS_msn=307769&_HLS_part=0", loader: XhrLoader, …}
```

--- a/src/controller/base-playlist-controller.ts
+++ b/src/controller/base-playlist-controller.ts
@@ -138,16 +138,18 @@ export default class BasePlaylistController implements NetworkComponentAPI {
       if (details.canBlockReload && details.endSN && details.advanced) {
         // Load level with LL-HLS delivery directives
         const lowLatencyMode = this.hls.config.lowLatencyMode;
+        const lastPartSn = details.lastPartSn;
+        const endSn = details.endSN;
         const lastPartIndex = details.lastPartIndex;
-        if (lowLatencyMode) {
-          msn = lastPartIndex !== -1 ? details.lastPartSn : details.endSN + 1;
-          part = lastPartIndex !== -1 ? lastPartIndex + 1 : undefined;
+        const hasParts = lastPartIndex !== -1;
+        const lastPart = lastPartSn === endSn;
+        // When low latency mode is disabled, we'll skip part requests once the last part index is found
+        const nextSnStartIndex = lowLatencyMode ? 0 : lastPartIndex;
+        if (hasParts) {
+          msn = lastPart ? endSn + 1 : lastPartSn;
+          part = lastPart ? nextSnStartIndex : lastPartIndex + 1;
         } else {
-          // This playlist update will be late by one part (0). There is no way to know the last part number,
-          // or request just the next sn without a part in most implementations.
-          msn =
-            lastPartIndex !== -1 ? details.lastPartSn + 1 : details.endSN + 1;
-          part = lastPartIndex !== -1 ? 0 : undefined;
+          msn = endSn + 1;
         }
         // Low-Latency CDN Tune-in: "age" header and time since load indicates we're behind by more than one part
         // Update directives to obtain the Playlist that has the estimated additional duration of media

